pkg/inventory/model/client.go

```diff
@@ -69,44 +68,29 @@ type Client struct {
 // Create the database.
 // Build the schema to support the specified models.
 // Optionally `purge` (delete) the DB first.
-func (r *Client) Open(purge bool) error {
+func (r *Client) Open(purge bool) (err error) {
 	if purge {
 		_ = os.Remove(r.path)
 	}
-	db, err := sql.Open("sqlite3", r.path)
+	r.db, err = sql.Open("sqlite3", r.path)
 	if err != nil {
 		r.log.V(3).Error(err, "open, failed.")
 		panic(err)
 	}
-	statements := []string{Pragma}
-	r.models = append(r.models, &Label{})
-	for _, m := range r.models {
-		ddl, err := Table{}.DDL(m)
-		if err != nil {
-			panic(err)
-		}
-		statements = append(statements, ddl...)
-	}
-	for _, ddl := range statements {
-		_, err = db.Exec(ddl)
+	defer func() {
 		if err != nil {
-			_ = db.Close()
-			return liberr.Wrap(
-				err,
-				"DDL failed.",
-				"ddl",
-				ddl)
+			_ = r.db.Close()
+			_ = os.Remove(r.path)
+			r.db = nil
 		}
-		r.log.V(4).Info(
-			"DDL executed.",
-			"ddl",
-			ddl)
+	}()
+	err = r.build()
+	if err != nil {
+		panic(err)
 	}
 
 	r.log.V(3).Info("DB opened.")
 
-	r.db = db
```

Comment:
The `Client` is accessed concurrently.  Setting the sqlite DB (field) to nil can result in SEGV.  I observed this while testing the frequent open/close cycles.